core/src/scene/sceneLoader.cpp

```diff
@@ -1049,6 +1024,67 @@ std::vector<StyleParam> SceneLoader::parseStyleParams(Node params, Scene& scene,
     return out;
 }
 
+StyleUniforms SceneLoader::parseStyleUniforms(const Node& value, Scene& scene) {
+    std::string type = "";
+    std::vector<UniformValue> uniformValues;
+    if (value.IsScalar()) { //float, int (bool), string (texture)
+        UniformValue uVal;
+        try {
+            uVal = value.as<float>();
+            type = "float";
+        } catch (const BadConversion& e) {
+            try {
+                uVal = value.as<bool>();
+                type = "bool";
+            } catch (const BadConversion& e) {
+                auto strVal = value.as<std::string>();
+                type = "sampler2D";
+                uVal = strVal;
+                auto texItr = scene.textures().find(strVal);
+                if (texItr == scene.textures().end()) {
+                    loadTexture(strVal, scene);
+                }
+            }
+        }
+        uniformValues.push_back(uVal);
+    } else if (value.IsSequence()) {
+        int size = value.size();
+        try {
+            type = "vec" + std::to_string(size);
+            UniformValue uVal;
+            switch (size) {
+                case 2:
+                    uVal = parseVec<glm::vec2>(value);
+                    break;
+                case 3:
+                    uVal = parseVec<glm::vec3>(value);
+                    break;
+                case 4:
+                    uVal = parseVec<glm::vec4>(value);
+                    break;
+                default:
+                    break;
+                    // TODO: Handle numeric arrays past 4 elements
+            }
+            uniformValues.push_back(uVal);
+        } catch (const BadConversion& e) { // array of strings (textures)
+            uniformValues.reserve(size);
+            type = "sampler2D";
+            for (const auto& strVal : value) {
```

Comment:
I'm unclear on the interpretation of an array of strings here, does it create an array of samplers?